cmd/snap/cmd_run.go

```diff
@@ -67,7 +67,7 @@ func (x *cmdRun) Execute(args []string) error {
 	if x.Hook != "" && x.Command != "" {
 		return fmt.Errorf("cannot use --hook and --command together")
 	}
-	if x.Revision != "" && x.Hook == "" {
+	if x.Revision != "unset" && x.Revision != "" && x.Hook == "" {
```

Comment:
No. `snap run` accepts the `-r` parameter for the revision, which previously defaulted to an empty string, meaning "no revision, run on the active one." That required some custom logic in the `HookManager` to see if the requested hook had a revision set, and if not, to pass an empty string. This PR updates `snap run` to also accept "unset" to mean "no revision, run on the active one," which means `HookManager` can simply use `snap.Revision.String()` without any custom logic, and things just work.